### PR TITLE
dock refactoring

### PR DESCRIPTION
**Pull request by Gustry**

### CODE REVIEW

**Comment by ismailsunni** [comment]
LGTM. Is it ready to merge?

**Comment by timlinux** [comment]
LGTM so far...

**Comment by Gustry** [comment]
Ok, finally this PR is ready to be reviewed.
We can still clean the dock and tests, but we can come back later, it's not my priority for the beta release. @ismailsunni if you are working on the wizard, I think you can check how I'm managing the error here. Let me know. Feel free to merge if it's ok for you.

File: safe/impact_function_v4/impact_function.py [hunk 1]
```diff
@@ -702,28 +743,84 @@ def debug_layer(self, layer, check_fields=True):
...
-        self._profiling_table = create_profile_layer(
-            self.performance_log_message())
-        _, name = self.debug_layer(self._profiling_table)
-        self._profiling_table = self.datastore.layer(name)
-        check_inasafe_fields(self._profiling_table)
+        """Run the whole impact function.
+
+        :return: A tuple with the status of the IF and an error message if
+            needed.
+            The status is 0 if everything was fine.
```
Comment (ismailsunni) [inline safe/impact_function_v4/impact_function.py]: Perhaps we can use a variable for the status code. For example:
SUCCESS = 1, INPUT_PROBLEM = 2, INASAFE_PROBLEM = 3 etc and put them like in definitions module.

Comment (Gustry) [inline safe/impact_function_v4/impact_function.py]: Yes. Feel free to do that in your PR or I will do that later ye. Thanks

**Pull request merged**